### ISSUE TITLE
Not able to reproduce results using the pre-trained weights

### ISSUE THREAD

**Issue by HashmatShadab** [MadryLab/smoothed-vit#4]
I am running the following script to evaluate the standard accuracy of ViT models

**ViT-S**
`python src/main.py  --dataset imagenet --data /path/to/imagenet/ --arch deit_small_patch16_224 --out-dir Pretrained_Models 
--exp-name vit_small  --resume --resume-ckpt-name deit_small_k19.pt --batch-size 64 --adv-train 0  --freeze-level -1  --eval-only 1  --adv-eval 0
`

**ViT-T**
`python src/main.py  --dataset imagenet --data /path/to/imagenet/ --arch deit_tiny_patch16_224 --out-dir Pretrained_Models 
--exp-name vit_tiny  --resume --resume-ckpt-name deit_tiny_k19.pt --batch-size 64 --adv-train 0  --freeze-level -1  --eval-only 1  --adv-eval 0
`

model weights at Pretrained_Models/vit_small/deit_small_k19.pt
model weights at Pretrained_Models/vit_tiny/deit_tiny_k19.pt

From Table 7 of the [paper](https://arxiv.org/abs/2110.07719)
ViT-S: Reported Accuracy - 63.5%   But got 69.14%
ViT-T: Reported Accuracy - 52.3%    But got 54.10%

Can you please provide some feedback on this?

**Comment by Hadisalman** [comment]
Hi @HashmatShadab! Using these commands, you are actually evaluating the standard accuracy of the `base model`, not the `smoothed model`.

To get the accuracies you pointed to from the table, you should be running evaluating the smoothed model, which can be done by running the certification code [here](https://github.com/MadryLab/smoothed-vit/tree/main#certifying-the-smoothed-classifier). In this script, the base model is converted into a smoothed model and evaluated for both standard and certified performance. Hope this helps!

**Comment by HashmatShadab** [comment]
Thanks for clarifying :)

**Comment by HashmatShadab** [comment]
## Evaluating Smooth model
I am trying to reproduce the results mentioned in Table 6 of the paper. The ablation size is fixed to b=4. The accuracies are evaluated on  adversarial patch sizes 2x2 and 4x4.


## ViT-T

python src/main.py 
      --dataset cifar10 
      --data /tmp 
      --arch deit_tiny_patch16_224 
      --out-dir cifar_models 
      --exp-name smooth 
      --batch-size 128 
      --adv-train 0 
      --freeze-level -1 
      --drop-tokens 
      --cifar-preprocess-type simple224 
      --resume 
      --eval-only 1 
      --certify 
      --certify-out-dir OUTDIR_CERT 
      --certify-mode col 
      --certify-ablation-size 4
      --certify-patch-size 2
      --resume-ckpt-name cifar10_vit-T.pt 

Output: acc: 0.8560, ablation 22.2275, smoothed 0.8503, certified 0.6862, delta: 16.0588, s: 4, m: 2
Reported Results: Certified accuracy 70.0%, Clean accuracy 85.5%

Output: acc: 0.8560, ablation 22.2275, smoothed 0.8503, certified 0.6061, delta: 16.0588, s: 4, m: 4
Reported Results: Certified accuracy 58.5%, Clean accuracy 85.5%

model weights at cifar_models/smooth/cifar10_vit-T.pt

Can you please let me know if there is anything I am missing, which is causing the variation, specifically in certified accuracy?
